File: public_web/tests/vixrex-matcher-parity.test.ts
import { describe, expect, it } from "vitest";
import fixtures from "../../shared/vixrex_motor_parity_fixtures.json";
import { resolveVixrexIntentMatches } from "../src/lib/vixrexIntentResolver";

type MatcherFixture = {
  id: string;
  input: string;
  expectedFieldKeys: string[];
  expectedMatchClasses: string[];
};

const matcherFixtures = (fixtures as { matcher: MatcherFixture[] }).matcher;

describe("Vixrex matcher shared parity fixtures — Next.js", () => {
  for (const fixture of matcherFixtures) {
    it(fixture.id, () => {
      const matches = resolveVixrexIntentMatches(fixture.input);
      expect(matches.map((match) => match.alan.anahtar)).toEqual(fixture.expectedFieldKeys);
      expect(matches.map((match) => match.matchClass)).toEqual(fixture.expectedMatchClasses);
    });
  }
});

File: test/vixrex_matcher_parity_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';
import 'package:vixrex/services/vixrex_nlu/vixrex_intent_resolver.dart';

void main() {
  final raw =
      jsonDecode(
            File('shared/vixrex_motor_parity_fixtures.json').readAsStringSync(),
          )
          as Map<String, dynamic>;
  final fixtures = (raw['matcher'] as List).cast<Map<String, dynamic>>();
  const resolver = VixrexIntentResolver();

  group('Vixrex matcher shared parity fixtures — Flutter', () {
    for (final fixture in fixtures) {
      test(fixture['id'] as String, () {
        final matches = resolver.resolveMatches(fixture['input'] as String);
        final expectedFields =
            (fixture['expectedFieldKeys'] as List).cast<String>();
        final expectedClasses =
            (fixture['expectedMatchClasses'] as List).cast<String>();

        expect(
          matches.map((match) => match.alan.anahtar).toList(),
          expectedFields,
        );
        expect(
          matches.map((match) => match.matchClass).toList(),
          expectedClasses,
        );
      });
    }
  });
}
